prow/cmd/phaino/local.go

```diff
@@ -270,42 +298,31 @@ func start(args []string) (*exec.Cmd, error) {
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	cmd.Stdin = os.Stdin
-
 	return cmd, cmd.Start()
 }
 
-func abort(ctx context.Context, log *logrus.Entry, cmd *exec.Cmd) error {
-	if err := cmd.Process.Signal(os.Interrupt); err != nil {
-		log.WithError(err).Warn("Interrupt error")
-	} else {
-		log.Warn("Interrupted")
-	}
-	ch := make(chan error)
-	go func() {
-		defer close(ch)
-		ch <- cmd.Wait()
-	}()
-	select {
-	case err := <-ch:
-		return err
-	case <-ctx.Done():
-		if err := cmd.Process.Kill(); err != nil { // TODO(fejta): docker rm
-			log.WithError(err).Warn("Kill error")
-			return err
-		}
-		log.Warn("Killed")
-	}
-	return nil
+func kill(cid, signal string) error {
+	cmd := exec.Command("docker", "kill", "--signal="+signal, cid)
+	cmd.Stdin = os.Stdin
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+var (
+	nameLock sync.Mutex
+	nameId   int
+)
+
+func containerID() string {
+	nameLock.Lock()
+	defer nameLock.Unlock()
+	nameId++
+	return fmt.Sprintf("phaino-%d-%d", os.Getpid(), nameId)
```

Comment:
It would be handy to include the `pj.Spec.Job` name in the container name so that it is easy to distinguish containers when multiple jobs are being run.